### ISSUE TITLE
FN S1541: no support for C# 9 patterns

### ISSUE THREAD

**Issue by dmitry-golovinov-sonarsource** [SonarSource/sonar-dotnet#4682]
### Description

Rule S1541 should support new C# 9 patterns

### Repro steps
```
    public bool PatternMatchingAnd(object arg) =>
        arg is true and true and true and true;         // FN

    public bool PatternMatchingOr(object arg) =>
        arg is true or true or true or true;            // FN
```
### Expected behavior

Issue should be raised if maximal complexity set to 3

### Actual behavior

Issue is not raised.

### Related information

* C#/VB.NET Plugins version 8.26
